src/codegen.cpp

```diff
@@ -5753,15 +5906,40 @@ static std::pair<std::unique_ptr<Module>, jl_llvm_functions_t>
     ctx.code = src->code;
 
     std::map<int, BasicBlock*> labels;
+    bool toplevel = false;
     ctx.module = jl_is_method(lam->def.method) ? lam->def.method->module : lam->def.module;
     ctx.linfo = lam;
+    ctx.name = name_from_method_instance(lam);
+    size_t nreq = 0;
+    int va = 0;
+    if (jl_is_method(lam->def.method)) {
+        ctx.nargs = nreq = lam->def.method->nargs;
+        jl_datatype_t *closure = jl_first_argument_datatype(lam->specTypes);
+        if (closure && jl_is_opaque_closure_type(closure)) {
+            // This is an opaque closure
+            ctx.is_opaque_closure = true;
+        }
```

Comment:
```suggestion
        // Check if we marked this an opaque_closure when it was created
        ctx.is_opaque_closure = lam->def.method->sig == jl_anytuple_type && lam->def.method->name == jl_symbol("opaque closure");
```

Reply:
I'll just make an explicit method field for it